crates/integration-tests/dns-ptr.lua

```diff
@@ -0,0 +1,28 @@
+local kumo = require 'kumo'
+
+local function contains(list, t)
+    for key, value in pairs(list) do
+        if value == t then
+            return true
+        end
+    end
+    return false
+end
+
```

Comment:
This test is technically dependent upon the local DNS service in terms of its test expectations, and also also the availability of functional network and DNS service.

To remove that dependency, we can put the resolver into a test mode where the zones are explicitly provided, as shown below.

A consequence of this is that the test case for `::2` below will no longer raise an error, but will return an empty table.

```suggestion
local zones = {
  [[
$ORIGIN 0.0.127.in-addr.arpa.
1 30 IN PTR localhost.
  ]],
  [[
$ORIGIN 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa.
@ 30 IN PTR localhost.
  ]],
}
kumo.dns.configure_test_resolver(zones)
```